File: src/main/java/rocks/sakira/sakurarosea/common/item/SakuraTeaItem.java
package rocks.sakira.sakurarosea.common.item;

import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import net.minecraft.potion.EffectInstance;
import net.minecraft.potion.Effects;
import net.minecraft.world.World;

public class SakuraTeaItem extends Item {
    public SakuraTeaItem(Properties properties) {
        super(properties);
    }

    public ItemStack onItemUseFinish(ItemStack stack, World worldIn, LivingEntity entityLiving) {
        entityLiving.addPotionEffect(new EffectInstance(Effects.SPEED, 8 * 20, 1));

        return entityLiving instanceof PlayerEntity && ((PlayerEntity) entityLiving).abilities.isCreativeMode ? stack : new ItemStack(Items.BOWL);
    }
}

File: src/main/kotlin/rocks/sakira/sakurarosea/item/SakuraTeaItem.kt
package rocks.sakira.sakurarosea.item

import net.minecraft.entity.LivingEntity
import net.minecraft.entity.effect.StatusEffectInstance
import net.minecraft.entity.effect.StatusEffects
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.item.Items
import net.minecraft.world.World

class SakuraTeaItem(settings: Settings) : Item(settings) {
    override fun finishUsing(stack: ItemStack, world: World, user: LivingEntity): ItemStack {
        user.addStatusEffect(StatusEffectInstance(StatusEffects.SPEED, 8 * 20, 1))

        return if (user is PlayerEntity && user.abilities.creativeMode) {
            stack
        } else {
            ItemStack(Items.BOWL)
        }
    }
}
